src/equations/bbm_bbm_1d.jl

```diff
@@ -69,7 +98,12 @@ function initial_condition_manufactured(x, t,
                                         mesh)
     eta = exp(t) * cospi(2 * (x - 2 * t))
     v = exp(t / 2) * sinpi(2 * (x - t / 2))
-    return SVector(eta, v)
+    if equations.bathymetry_type isa BathymetryFlat
+        D = 4
+    else # equations.bathymetry_type isa BathymetryVariable
+        D = 5 + 2 * cospi(2 * x)
+    end
```

Comment:
This is a (minor) type instability in the `if` block. Since we just use `Float64` everywhere, we could fix it via

```suggestion
    if equations.bathymetry_type isa BathymetryFlat
        D = 4.0
    else # equations.bathymetry_type isa BathymetryVariable
        D = 5 + 2 * cospi(2 * x)
    end
```